src/Native/System.Native/pal_mount.cpp

```diff
@@ -134,7 +141,7 @@ SystemNative_GetFormatInfoForMountPoint(const char* name, char* formatNameBuffer
         }
 #else
         assert(formatType != nullptr);
-        *formatType = stats.f_type;
+        *formatType = GetFTypeAsInt64(stats.f_type);
```

Comment:
Makes sense. I was trying to simplify the cmake check which I had before and ended up using typedefs instead of simple static_cast. Also, didn't knew about SignedCast helper. I have now used that instead and amended the commit. Thank you! :)